tools/nodejs_api/src_cpp/all_async_worker.cpp

```diff
@@ -0,0 +1,43 @@
+#include "include/all_async_worker.h"
+
+#include <chrono>
+#include <thread>
+#include <vector>
+
+#include "include/node_query_result.h"
+#include "include/util.h"
+
+using namespace std;
+
+AllAsyncWorker::AllAsyncWorker(Function& callback, shared_ptr<kuzu::main::QueryResult>& queryResult)
+    : AsyncWorker(callback), queryResult(queryResult) {};
+
+void AllAsyncWorker::Execute() {
+    try {
+        size_t i = 0;
+        while (this->queryResult->hasNext()) {
+            auto row = this->queryResult->getNext();
+            allResult.emplace_back(vector<unique_ptr<kuzu::common::Value>>(row->len()));
+            for (size_t j = 0; j < row->len(); j++) {
+                allResult[i][j] =
+                    make_unique<kuzu::common::Value>(kuzu::common::Value(*row->getValue(j)));
+            }
+            i++;
+        }
+    } catch(const std::exception &exc) {
+        SetError("Unsuccessful async All/Each callback: " + std::string(exc.what()));
+    }
+};
+
+void AllAsyncWorker::OnOK() {
+    Napi::Array arr = Napi::Array::New(Env(), allResult.size());
+    for (size_t i=0; i < allResult.size(); i++) {
```

Comment:
I think it is actually possible to move this work from the main thread with `ThreadSafeFunction` (https://github.com/nodejs/node-addon-api/blob/main/doc/threadsafe_function.md). Please consider using it instead of performing the conversions in `OnOK`.